include/picongpu/plugins/openPMD/restart/RestartFieldLoader.hpp

```diff
@@ -0,0 +1,229 @@
+            DataSpace< simDim > local_domain_size =
+                params->window.localDimensions.size;
+            bool useLinearIdxAsDestination = false;
+
+            /* Patch for non-domain-bound fields
+             * This is an ugly fix to allow output of reduced 1d PML buffers,
+             * that are the same size on each domain.
+             */
+            if( !isDomainBound )
+            {
+                auto const field_layout = params->gridLayout;
+                auto const field_no_guard =
+                    field_layout.getDataSpaceWithoutGuarding();
+                auto const elementCount = field_no_guard.productOfComponents();
+                auto const & gridController =
+                    Environment< simDim >::get().GridController();
+                auto const rank = gridController.getGlobalRank();
+                domain_offset = DataSpace< simDim >::create( 0 );
+                domain_offset[ 0 ] = rank * elementCount;
+                local_domain_size = DataSpace< simDim >::create( 1 );
+                local_domain_size[ 0 ] = elementCount;
+                useLinearIdxAsDestination = true;
+            }
+
+            ::openPMD::Series & series = *params->openPMDSeries;
+            ::openPMD::Container<::openPMD::Mesh > & meshes =
+                series.iterations[ params->currentStep ].meshes;
+
+            auto destBox = field.getHostBuffer().getDataBox();
+            for( uint32_t n = 0; n < numComponents; ++n )
+            {
+                // Read the subdomain which belongs to our mpi position.
+                // The total grid size must match the grid size of the stored
+                // data.
+                log< picLog::INPUT_OUTPUT >(
+                    "openPMD: Read from domain: offset=%1% size=%2%" ) %
+                    domain_offset % local_domain_size;
+                ::openPMD::RecordComponent rc = numComponents > 1
+                    ? meshes[ objectName ][ name_lookup_tpl[ n ] ]
+                    : meshes[ objectName ][::openPMD::RecordComponent::SCALAR ];
+
+                log< picLog::INPUT_OUTPUT >(
+                    "openPMD: Read from field '%1%'" ) %
+                    objectName;
+
+                auto ndim = rc.getDimensionality();
+                ::openPMD::Offset start = asStandardVector<
+                    DataSpace< simDim > &,
+                    ::openPMD::Offset >( domain_offset );
+                ::openPMD::Extent count = asStandardVector<
+                    DataSpace< simDim > &,
+                    ::openPMD::Extent >( local_domain_size );
+
+                log< picLog::INPUT_OUTPUT >(
+                    "openPMD: Allocate %1% elements" ) %
+                    local_domain_size.productOfComponents();
+
+                /// \todo float_X should be some kind of gridBuffer's
+                /// GetComponentsType<ValueType>::type
+                /* magic parameters (0, 1): `from_step` (not used in streams),
```

Comment:
I do not understand this comment and the todo above

Reply:
Again a case of copy-pasting from ADIOS. I don't know if the todo is still relevant, I've moved it back to the right position again. The other thing was Adios-specific.